internal/manifests/memberlist/gossip.go

```diff
@@ -46,3 +51,55 @@ func BuildGossip(tempo v1alpha1.TempoStack) *corev1.Service {
 		},
 	}
 }
+
+func resetEnvVar(podSpec *corev1.PodSpec, name string) {
```

Comment:
I'm not sure why we need this? We don't let users set any custom env vars, only the operator can set them. So afaics it's impossible that the env var is already set and needs to be unset.